### DISCUSSION TITLE
Does OpenDDS for Java have performance test?

### DISCUSSION THREAD

**Discussion by yuan-ge** [OpenDDS/OpenDDS#4850] [Q&A]
These days I'm care about the latency when I use OpenDDS for Java, and I did some tests. The loopback latency of OpenDDS Java with 100 bytes is about 700us and latency on two machines is about 1.7ms. Is the performance normal?
Besides, I didn't find performance tests in java dir. Has anyone tested the performance with java?
Thanks!

**Comment by mitza-oci** [comment]
What's the configuration you're testing?  How does it compare to https://opendds.org/perf/cent210803/latency_results.html ?

In general I wouldn't expect the Java performance to be much different than for C++.  There isn't really an "OpenDDS for Java", there's just OpenDDS (the very same middleware libraries) with a Java binding on top.  

The one exception would be a case where the overhead of copying data between the JVM's representation and the actual OpenDDS representation is costly.

**Reply by yuan-ge** [reply]
After further testing, I have some speculations. I think it may be the overhead of JVM itself and data copying from the middleware to the JVM. By replacing different hardware devices, I found that there are obvious differences in latency. The configuration I use is rtps-udp, and the configurations of the two machines are PC and Raspberry Pi. Later, I tested the local loopback latency on the Raspberry Pi and found that it is as high as 3.8 milliseconds.
